### ISSUE TITLE
Is it worth it to include floating mesh normal checks?

### ISSUE THREAD

**Issue by albertoferna** [RubendeBruin/DAVE#34]
While doing some test in DAVE, I found something that is not really an issue. Flotation is wrongly calculated if normals of the mesh are inconsistent. I imported a mesh file from a conversion and it turned out that it was not correctly "watertight". Displacement in that case was negative.
From my perspective it was very simple to find, but it may be an issue for other people. It may be worth it to include some check in DAVE?
I know it can be done in a million other software so that when the mesh is imported it is correct. Just a thought

**Comment by RubendeBruin** [comment]
Hi @albertoferna . Thanks for sharing your ideas! The mesh import is done via VTK and I see that vtk has some routines as well for checking normal consistency. Not sure how to test for weathertightness other than evaluating the integral in different directions and comparing the results.
I was also thinking about including an option for inverting the normals which may be useful when working with contact shapes.

**Comment by albertoferna** [comment]
To be honest, I use vtk through a wrapper: pyvista. But, I believe that vtk.vtkFeatureEdges() could help there. It can be called to extract only boundary edges. Obviously, a closed body does not have boundary edges.
I don't know the function, but I seem to remember that there was one also to make normals consistent.

**Comment by RubendeBruin** [comment]
Hi @albertoferna , sorry that it took so long. 

I've added checks on normals as well as boundary and non-manifold edges. 
vtk.vtkFeatureEdges() was a good idea but in the end I just implemented it directly using numpy.

Also new is the option to invert nomals
![image](https://user-images.githubusercontent.com/34062862/111171758-81e07100-85a5-11eb-8c63-09a3f067d0e3.png)

![image](https://user-images.githubusercontent.com/34062862/111172009-bbb17780-85a5-11eb-9ff9-f66083265845.png)
